File: 1071. Greatest Common Divisor of Strings/main.cpp
#include <numeric>
#include <string>

using namespace std;

class Solution {
    size_t gcd(size_t a, size_t b) {
        return (b == 0) ? a : gcd(b, a % b);
    }

public:
    string gcdOfStrings(string str1, string str2) {
        string result;

        if (str1 + str2 != str2 + str1) {
            return result;
        }

        size_t gcdLen = gcd(str1.size(), str2.size());
        result = str1.substr(0, gcdLen);

        return result;
    }
};

class Solution {
public:
    string gcdOfStrings(string str1, string str2) {
        string result;

        if (str1 + str2 != str2 + str1) {
            return result;
        }

        size_t gcdLen = gcd(str1.size(), str2.size());
        result = str1.substr(0, gcdLen);

        return result;
    }
};

class Solution {
public:
    string gcdOfStrings(string str1, string str2) {
        if (str1 + str2 != str2 + str1) {
            return "";
        }

        return str1.substr(0, gcd(str1.size(), str2.size()));
    }
};

File: 1071. Greatest Common Divisor of Strings/main.py
import math


class Solution:
    def gcdOfStrings(self, str1: str, str2: str) -> str:
        if str1 + str2 != str2 + str1:
            return ""

        return str1[: math.gcd(len(str1), len(str2))]


class Solution:
    def gcdOfStrings(self, str1: str, str2: str) -> str:
        def gcd(a: int, b: int) -> int:
            if b == 0:
                return a

            return gcd(b, a % b)

        if str1 + str2 != str2 + str1:
            return ""

        return str1[: gcd(len(str1), len(str2))]
